File: ios/Sasquatsh/Views/Toolbox/RoundCounterView.swift
import SwiftUI

struct RoundCounterView: View {
    @Environment(\.dismiss) private var dismiss
    @State private var vm = RoundCounterViewModel()
    @State private var showSettings = false

    private let durationOptions = [30, 60, 90, 120, 180, 300, 600]

    var body: some View {
        GeometryReader { geo in
            let compact = geo.size.height < 700

            VStack(spacing: 0) {
                // Compact nav + timer
                CompactNavBar(title: "Round Counter") { dismiss() }
                timerBar(compact: compact)

                Spacer(minLength: 0)

                // Center: round number + controls — always visible
                VStack(spacing: compact ? 12 : 20) {
                    // Round label
                    if vm.isCountdownMode {
                        Text("ROUND \(vm.state.roundNumber) OF \(vm.maxRounds)")
                            .font(.system(size: 13, weight: .bold))
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                            .tracking(3)
                    } else {
                        Text("ROUND")
                            .font(.system(size: 13, weight: .bold))
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                            .tracking(3)
                    }

                    // Big round number
                    Button {
                        vm.beginEditRound()
                    } label: {
                        ZStack {
                            // Glow ring
                            Circle()
                                .stroke(
                                    vm.reachedMaxRound ? Color.md3Error.opacity(0.3) :
                                    vm.isTimerRunning ? Color.md3Primary.opacity(0.15) :
                                    Color.md3OnSurfaceVariant.opacity(0.08),
                                    lineWidth: 6
                                )
                                .frame(width: compact ? 160 : 200, height: compact ? 160 : 200)

                            // Progress ring (countdown mode)
                            if vm.isCountdownMode && vm.roundDurationSeconds > 0 {
                                Circle()
                                    .trim(from: 0, to: max(0, vm.countdownRemaining / Double(vm.roundDurationSeconds)))
                                    .stroke(
                                        vm.countdownRemaining <= 10 && vm.isTimerRunning ? Color.md3Error : Color.md3Primary,
                                        style: StrokeStyle(lineWidth: 6, lineCap: .round)
                                    )
                                    .frame(width: compact ? 160 : 200, height: compact ? 160 : 200)
                                    .rotationEffect(.degrees(-90))
                                    .animation(.linear(duration: 0.1), value: vm.countdownRemaining)
                            }

                            VStack(spacing: 2) {
                                Text("\(vm.state.roundNumber)")
                                    .font(.system(size: compact ? 64 : 80, weight: .bold, design: .rounded))
                                    .foregroundStyle(vm.reachedMaxRound ? Color.md3Error : Color.md3OnSurface)
                                    .contentTransition(.numericText())

                                if vm.isCountdownMode {
                                    Text(vm.formattedCountdown)
                                        .font(.system(size: 16, weight: .medium, design: .monospaced))
                                        .foregroundStyle(vm.countdownRemaining <= 10 && vm.isTimerRunning ? Color.md3Error : Color.md3Primary)
                                }
                            }
                        }
                    }
                    .accessibilityLabel("Round \(vm.state.roundNumber)")

                    if vm.reachedMaxRound {
                        Text("Final round!")
                            .font(.md3LabelLarge)
                            .foregroundStyle(Color.md3Error)
                            .transition(.scale.combined(with: .opacity))
                    }

                    // +/- controls
                    HStack(spacing: 40) {
                        Button {
                            withAnimation(.spring(response: 0.3, dampingFraction: 0.6)) {
                                vm.decrement()
                            }
                        } label: {
                            Image(systemName: "minus")
                                .font(.system(size: 26, weight: .semibold))
                                .frame(width: 64, height: 64)
                                .foregroundStyle(vm.state.roundNumber <= vm.state.minimumRound ? Color.md3OnSurfaceVariant.opacity(0.3) : Color.md3OnSecondaryContainer)
                                .background(vm.state.roundNumber <= vm.state.minimumRound ? Color.md3SurfaceContainerHigh : Color.md3SecondaryContainer)
                                .clipShape(Circle())
                        }
                        .disabled(vm.state.roundNumber <= vm.state.minimumRound)

                        // Play/pause in the middle
                        Button {
                            vm.toggleTimer()
                        } label: {
                            Image(systemName: vm.isTimerRunning ? "pause.fill" : "play.fill")
                                .font(.system(size: 22))
                                .frame(width: 56, height: 56)
                                .foregroundStyle(Color.md3OnPrimary)
                                .background(Color.md3Primary)
                                .clipShape(Circle())
                                .shadow(color: Color.md3Primary.opacity(0.3), radius: 8, y: 4)
                        }

                        Button {
                            withAnimation(.spring(response: 0.3, dampingFraction: 0.6)) {
                                vm.increment()
                            }
                        } label: {
                            Image(systemName: "plus")
                                .font(.system(size: 26, weight: .semibold))
                                .frame(width: 64, height: 64)
                                .foregroundStyle(Color.md3OnPrimary)
                                .background(Color.md3Primary)
                                .clipShape(Circle())
                        }
                    }
                }

                Spacer(minLength: 0)

                // Bottom bar
                HStack(spacing: 12) {
                    Button {
                        vm.showResetConfirmation = true
                    } label: {
                        HStack(spacing: 6) {
                            Image(systemName: "arrow.counterclockwise")
                            Text("Reset")
                        }
                        .outlinedButtonStyle()
                    }

                    Button {
                        showSettings = true
                    } label: {
                        Image(systemName: "gearshape")
                            .font(.md3LabelLarge)
                            .frame(width: 48, height: 40)
                            .background(Color.md3SecondaryContainer)
                            .foregroundStyle(Color.md3OnSecondaryContainer)
                            .clipShape(Capsule())
                    }

                    Button {
                        vm.newSession()
                    } label: {
                        HStack(spacing: 6) {
                            Image(systemName: "plus.circle")
                            Text("New")
                        }
                        .primaryButtonStyle()
                    }
                }
                .padding(.horizontal, 20)
                .padding(.vertical, 10)
                .background(Color.md3SurfaceContainer)
            }
        }
        .background(Color.md3SurfaceContainer)
        .toolbar(.hidden, for: .navigationBar)
        .task {
            await vm.loadSaved()
        }
        .onAppear {
            UIApplication.shared.isIdleTimerDisabled = true
        }
        .onDisappear {
            UIApplication.shared.isIdleTimerDisabled = false
        }
        .alert("Reset Round?", isPresented: $vm.showResetConfirmation) {
            Button("Cancel", role: .cancel) {}
            Button("Reset", role: .destructive) {
                withAnimation { vm.reset() }
            }
        } message: {
            Text("Reset to round \(vm.state.startingRound) and clear timers?")
        }
        .alert("Set Round", isPresented: $vm.showEditRound) {
            TextField("Round", text: $vm.editRoundText)
                .keyboardType(.numberPad)
            Button("Cancel", role: .cancel) {}
            Button("Set") { vm.commitEditRound() }
        } message: {
            Text("Enter the round number.")
        }
        .sheet(isPresented: $showSettings) {
            settingsSheet
        }
    }

    // MARK: - Timer Bar

    private func timerBar(compact: Bool) -> some View {
        HStack(spacing: 12) {
            // Round / countdown time
            VStack(spacing: 1) {
                Text(vm.isCountdownMode ? "Left" : "Round")
                    .font(.system(size: 11, weight: .medium))
                    .foregroundStyle(Color.md3OnSurfaceVariant)
                Text(vm.isCountdownMode ? vm.formattedCountdown : vm.formattedRoundTime)
                    .font(.system(size: 20, weight: .semibold, design: .monospaced))
                    .foregroundStyle(
                        vm.isCountdownMode && vm.countdownRemaining <= 10 && vm.isTimerRunning
                            ? Color.md3Error : Color.md3OnSurface
                    )
                    .contentTransition(.numericText())
            }

            Divider().frame(height: 28)

            // Session time
            VStack(spacing: 1) {
                Text("Session")
                    .font(.system(size: 11, weight: .medium))
                    .foregroundStyle(Color.md3OnSurfaceVariant)
                Text(vm.formattedSessionTime)
                    .font(.system(size: 20, weight: .semibold, design: .monospaced))
                    .foregroundStyle(Color.md3OnSurface)
                    .contentTransition(.numericText())
            }

            if vm.isTimerRunning {
                Circle()
                    .fill(Color.md3Error)
                    .frame(width: 8, height: 8)
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 10)
        .padding(.horizontal, 20)
        .background(Color.md3Surface)
    }

    // MARK: - Settings Sheet

    private var settingsSheet: some View {
        NavigationStack {
            List {
                Section {
                    Toggle(isOn: $vm.isCountdownMode) {
                        Label("Countdown Mode", systemImage: "timer")
                    }
                    .tint(Color.md3Primary)
                }

                if vm.isCountdownMode {
                    Section("Round Duration") {
                        ForEach(durationOptions, id: \.self) { secs in
                            Button {
                                vm.roundDurationSeconds = secs
                            } label: {
                                HStack {
                                    Text(durationLabel(secs))
                                        .foregroundStyle(Color.md3OnSurface)
                                    Spacer()
                                    if vm.roundDurationSeconds == secs {
                                        Image(systemName: "checkmark")
                                            .foregroundStyle(Color.md3Primary)
                                    }
                                }
                            }
                        }
                    }

                    Section("Max Rounds") {
                        HStack {
                            Text("Rounds")
                            Spacer()
                            HStack(spacing: 12) {
                                Button {
                                    if vm.maxRounds > 1 { vm.maxRounds -= 1 }
                                } label: {
                                    Image(systemName: "minus.circle.fill")
                                        .font(.system(size: 24))
                                        .foregroundStyle(Color.md3Primary)
                                }
                                .disabled(vm.maxRounds <= 1)

                                Text("\(vm.maxRounds)")
                                    .font(.system(size: 22, weight: .bold, design: .rounded))
                                    .frame(width: 40)

                                Button {
                                    if vm.maxRounds < 99 { vm.maxRounds += 1 }
                                } label: {
                                    Image(systemName: "plus.circle.fill")
                                        .font(.system(size: 24))
                                        .foregroundStyle(Color.md3Primary)
                                }
                                .disabled(vm.maxRounds >= 99)
                            }
                        }
                    }
                }
            }
            .navigationTitle("Settings")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .confirmationAction) {
                    Button("Done") { showSettings = false }
                }
            }
        }
        .presentationDetents([.medium])
    }

    // MARK: - Helpers

    private func durationLabel(_ seconds: Int) -> String {
        if seconds < 60 { return "\(seconds) seconds" }
        let mins = seconds / 60
        return mins == 1 ? "1 minute" : "\(mins) minutes"
    }

    private func iconForEvent(_ type: String) -> String {
        switch type {
        case "incrementRound": return "arrow.up"
        case "decrementRound": return "arrow.down"
        case "setRound": return "pencil"
        case "resetRound": return "arrow.counterclockwise"
        case "autoIncrement": return "arrow.up.circle"
        case "maxRoundReached": return "flag.checkered"
        default: return "circle"
        }
    }

    private func labelForEvent(_ event: RoundCounterEvent) -> String {
        switch event.eventType {
        case "incrementRound": return "Round \(event.roundNumber)"
        case "decrementRound": return "Round \(event.roundNumber)"
        case "setRound": return "Set to round \(event.roundNumber)"
        case "resetRound": return "Reset to round \(event.roundNumber)"
        case "autoIncrement": return "Auto → Round \(event.roundNumber)"
        case "maxRoundReached": return "Max round \(event.roundNumber) reached"
        default: return event.eventType
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/toolbox/RoundCounterView.kt
package com.sasquatsh.app.views.toolbox

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Pause
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FloatingActionButton
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.drawBehind
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import com.sasquatsh.app.viewmodels.RoundCounterViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RoundCounterView(
    onBack: () -> Unit = {},
    viewModel: RoundCounterViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsState()
    var showSettings by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            TopAppBar(title = { Text("Round Counter") })
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
        ) {
            // Timer bar
            TimerBar(uiState)

            Spacer(modifier = Modifier.weight(1f))

            // Center: round number + controls
            Column(
                modifier = Modifier.fillMaxWidth(),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Round label
                if (uiState.isCountdownMode) {
                    Text(
                        "ROUND ${uiState.state.roundNumber} OF ${uiState.maxRounds}",
                        fontSize = 13.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        letterSpacing = 3.sp
                    )
                } else {
                    Text(
                        "ROUND",
                        fontSize = 13.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        letterSpacing = 3.sp
                    )
                }

                Spacer(modifier = Modifier.height(20.dp))

                // Big round number with progress ring
                val ringColor = when {
                    uiState.reachedMaxRound -> MaterialTheme.colorScheme.error.copy(alpha = 0.3f)
                    uiState.isTimerRunning -> MaterialTheme.colorScheme.primary.copy(alpha = 0.15f)
                    else -> MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.08f)
                }
                val progressColor = if (uiState.countdownRemaining <= 10000 && uiState.isTimerRunning)
                    MaterialTheme.colorScheme.error else MaterialTheme.colorScheme.primary

                Box(
                    modifier = Modifier
                        .size(200.dp)
                        .drawBehind {
                            // Background ring
                            drawCircle(
                                color = ringColor,
                                radius = size.minDimension / 2,
                                style = Stroke(width = 6.dp.toPx())
                            )
                            // Progress ring (countdown mode)
                            if (uiState.isCountdownMode && uiState.roundDurationSeconds > 0) {
                                val progress =
                                    (uiState.countdownRemaining.toFloat() / (uiState.roundDurationSeconds * 1000f)).coerceIn(0f, 1f)
                                drawArc(
                                    color = progressColor,
                                    startAngle = -90f,
                                    sweepAngle = progress * 360f,
                                    useCenter = false,
                                    topLeft = Offset.Zero,
                                    size = Size(size.width, size.height),
                                    style = Stroke(
                                        width = 6.dp.toPx(),
                                        cap = StrokeCap.Round
                                    )
                                )
                            }
                        }
                        .clickable { viewModel.beginEditRound() },
                    contentAlignment = Alignment.Center
                ) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text(
                            "${uiState.state.roundNumber}",
                            fontSize = 80.sp,
                            fontWeight = FontWeight.Bold,
                            color = if (uiState.reachedMaxRound) MaterialTheme.colorScheme.error
                            else MaterialTheme.colorScheme.onSurface
                        )
                        if (uiState.isCountdownMode) {
                            Text(
                                uiState.formattedCountdown,
                                fontSize = 16.sp,
                                fontWeight = FontWeight.Medium,
                                color = if (uiState.countdownRemaining <= 10000 && uiState.isTimerRunning)
                                    MaterialTheme.colorScheme.error else MaterialTheme.colorScheme.primary
                            )
                        }
                    }
                }

                if (uiState.reachedMaxRound) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        "Final round!",
                        style = MaterialTheme.typography.labelLarge,
                        color = MaterialTheme.colorScheme.error
                    )
                }

                Spacer(modifier = Modifier.height(20.dp))

                // +/- controls
                Row(
                    horizontalArrangement = Arrangement.spacedBy(40.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    // Minus
                    FloatingActionButton(
                        onClick = { viewModel.decrement() },
                        modifier = Modifier.size(64.dp),
                        containerColor = if (uiState.state.roundNumber <= uiState.state.minimumRound)
                            MaterialTheme.colorScheme.surfaceContainerHigh
                        else MaterialTheme.colorScheme.secondaryContainer,
                        shape = CircleShape
                    ) {
                        Icon(
                            Icons.Default.Remove,
                            contentDescription = "Decrease",
                            modifier = Modifier.size(26.dp),
                            tint = if (uiState.state.roundNumber <= uiState.state.minimumRound)
                                MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.3f)
                            else MaterialTheme.colorScheme.onSecondaryContainer
                        )
                    }

                    // Play/Pause
                    FloatingActionButton(
                        onClick = { viewModel.toggleTimer() },
                        modifier = Modifier.size(56.dp),
                        containerColor = MaterialTheme.colorScheme.primary,
                        shape = CircleShape
                    ) {
                        Icon(
                            if (uiState.isTimerRunning) Icons.Default.Pause else Icons.Default.PlayArrow,
                            contentDescription = if (uiState.isTimerRunning) "Pause" else "Play",
                            modifier = Modifier.size(22.dp),
                            tint = MaterialTheme.colorScheme.onPrimary
                        )
                    }

                    // Plus
                    FloatingActionButton(
                        onClick = { viewModel.increment() },
                        modifier = Modifier.size(64.dp),
                        containerColor = MaterialTheme.colorScheme.primary,
                        shape = CircleShape
                    ) {
                        Icon(
                            Icons.Default.Add,
                            contentDescription = "Increase",
                            modifier = Modifier.size(26.dp),
                            tint = MaterialTheme.colorScheme.onPrimary
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.weight(1f))

            // Bottom bar
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 10.dp),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                OutlinedButton(
                    onClick = { viewModel.showResetConfirmation(true) },
                    modifier = Modifier.weight(1f)
                ) {
                    Icon(Icons.Default.Refresh, contentDescription = null, modifier = Modifier.size(16.dp))
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Reset")
                }

                IconButton(
                    onClick = { showSettings = true }
                ) {
                    Icon(Icons.Default.Settings, contentDescription = "Settings")
                }

                Button(
                    onClick = { viewModel.newSession() },
                    modifier = Modifier.weight(1f)
                ) {
                    Icon(Icons.Default.Add, contentDescription = null, modifier = Modifier.size(16.dp))
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("New")
                }
            }
        }
    }

    // Reset confirmation dialog
    if (uiState.showResetConfirmation) {
        AlertDialog(
            onDismissRequest = { viewModel.showResetConfirmation(false) },
            title = { Text("Reset Round?") },
            text = { Text("Reset to round ${uiState.state.startingRound} and clear timers?") },
            confirmButton = {
                TextButton(onClick = {
                    viewModel.reset()
                    viewModel.showResetConfirmation(false)
                }) {
                    Text("Reset", color = MaterialTheme.colorScheme.error)
                }
            },
            dismissButton = {
                TextButton(onClick = { viewModel.showResetConfirmation(false) }) {
                    Text("Cancel")
                }
            }
        )
    }

    // Edit round dialog
    if (uiState.showEditRound) {
        AlertDialog(
            onDismissRequest = { viewModel.commitEditRound() },
            title = { Text("Set Round") },
            text = {
                OutlinedTextField(
                    value = uiState.editRoundText,
                    onValueChange = { viewModel.updateEditRoundText(it) },
                    label = { Text("Round") },
                    singleLine = true
                )
            },
            confirmButton = {
                TextButton(onClick = { viewModel.commitEditRound() }) {
                    Text("Set")
                }
            },
            dismissButton = {
                TextButton(onClick = { viewModel.commitEditRound() }) {
                    Text("Cancel")
                }
            }
        )
    }

    // Settings sheet
    if (showSettings) {
        SettingsSheet(
            uiState = uiState,
            viewModel = viewModel,
            onDismiss = { showSettings = false }
        )
    }
}

@Composable
private fun TimerBar(uiState: com.sasquatsh.app.viewmodels.RoundCounterUiState) {
    Surface(
        color = MaterialTheme.colorScheme.surface,
        modifier = Modifier.fillMaxWidth()
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 10.dp, horizontal = 20.dp),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(
                    if (uiState.isCountdownMode) "Left" else "Round",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Medium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    if (uiState.isCountdownMode) uiState.formattedCountdown else uiState.formattedRoundTime,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = if (uiState.isCountdownMode && uiState.countdownRemaining <= 10000 && uiState.isTimerRunning)
                        MaterialTheme.colorScheme.error else MaterialTheme.colorScheme.onSurface
                )
            }

            Spacer(modifier = Modifier.width(16.dp))

            // Divider
            Box(
                modifier = Modifier
                    .height(28.dp)
                    .width(1.dp)
                    .drawBehind {
                        drawLine(
                            color = Color.Gray.copy(alpha = 0.3f),
                            start = Offset(0f, 0f),
                            end = Offset(0f, size.height)
                        )
                    }
            )

            Spacer(modifier = Modifier.width(16.dp))

            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(
                    "Session",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Medium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    uiState.formattedSessionTime,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }

            if (uiState.isTimerRunning) {
                Spacer(modifier = Modifier.width(8.dp))
                Box(
                    modifier = Modifier
                        .size(8.dp)
                        .drawBehind {
                            drawCircle(color = Color.Red)
                        }
                )
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun SettingsSheet(
    uiState: com.sasquatsh.app.viewmodels.RoundCounterUiState,
    viewModel: RoundCounterViewModel,
    onDismiss: () -> Unit
) {
    val sheetState = rememberModalBottomSheetState()

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(
                "Settings",
                style = MaterialTheme.typography.titleLarge,
                modifier = Modifier.padding(bottom = 16.dp)
            )

            // Countdown mode toggle
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text("Countdown Mode", style = MaterialTheme.typography.bodyLarge)
                Switch(
                    checked = uiState.isCountdownMode,
                    onCheckedChange = { viewModel.updateCountdownMode(it) }
                )
            }

            if (uiState.isCountdownMode) {
                Spacer(modifier = Modifier.height(16.dp))
                Text("Round Duration", style = MaterialTheme.typography.titleSmall)
                Spacer(modifier = Modifier.height(8.dp))

                val durationOptions = listOf(30, 60, 90, 120, 180, 300, 600)
                durationOptions.forEach { secs ->
                    val label = if (secs < 60) "$secs seconds"
                    else "${secs / 60} minute${if (secs / 60 == 1) "" else "s"}"

                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clickable { viewModel.updateRoundDurationSeconds(secs) }
                            .padding(vertical = 8.dp),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text(label, style = MaterialTheme.typography.bodyMedium)
                        if (uiState.roundDurationSeconds == secs) {
                            Text(
                                "Selected",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.primary
                            )
                        }
                    }
                }

                Spacer(modifier = Modifier.height(16.dp))
                Text("Max Rounds", style = MaterialTheme.typography.titleSmall)
                Spacer(modifier = Modifier.height(8.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("Rounds", style = MaterialTheme.typography.bodyMedium)
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(12.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        IconButton(
                            onClick = { if (uiState.maxRounds > 1) viewModel.updateMaxRounds(uiState.maxRounds - 1) },
                            enabled = uiState.maxRounds > 1
                        ) {
                            Icon(Icons.Default.Remove, contentDescription = "Decrease")
                        }
                        Text(
                            "${uiState.maxRounds}",
                            fontSize = 22.sp,
                            fontWeight = FontWeight.Bold
                        )
                        IconButton(
                            onClick = { if (uiState.maxRounds < 99) viewModel.updateMaxRounds(uiState.maxRounds + 1) },
                            enabled = uiState.maxRounds < 99
                        ) {
                            Icon(Icons.Default.Add, contentDescription = "Increase")
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(16.dp))
            Button(
                onClick = onDismiss,
                modifier = Modifier.fillMaxWidth()
            ) {
                Text("Done")
            }
            Spacer(modifier = Modifier.height(16.dp))
        }
    }
}
